rust/agama-lib/share/profile.schema.json

```diff
@@ -224,54 +219,314 @@
+                                "title": "Whether Btrfs snapshots should be configured",
+                                "type": "boolean"
+                              }
+                            }
+                          }
+                        },
+                        "required": ["btrfs"]
+                      }
+                    ]
+                  },
+                  "size": {
+                    "title": "Size limits",
+                    "oneOf": [
+                      { "const": "auto" },
+                      { "$ref": "#/$defs/sizeString" },
+                      { "$ref": "#/$defs/sizeInteger" },
+                      {
+                        "description": "Lower size limit and optionally upper size limit",
+                        "type": "array",
+                        "items": {
+                          "$ref": "#/$defs/sizeValue"
+                        },
+                        "minItems": 1,
+                        "maxItems": 2
+                      },
+                      {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": {
+                          "min": {
+                            "title": "Mandatory lower size limit",
+                            "$ref": "#/$defs/sizeValue"
+                          },
+                          "max": {
+                            "title": "Optional upper size limit",
+                            "$ref": "#/$defs/sizeValue"
+                          }
+                        },
+                        "required": ["min"]
+                      }
+                    ]
+                  },
+                  "target": {
+                    "title": "Location of the resulting file system",
+                    "oneOf": [
+                      {
+                        "const": "default"
+                      },
+                      {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": { "newPartition": { "type": "string" } },
+                        "required": ["newPartition"]
+                      },
+                      {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": { "newVg": { "type": "string" } },
+                        "required": ["newVg"]
+                      },
+                      {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": { "device": { "type": "string" } },
+                        "required": ["device"]
+                      },
+                      {
+                        "type": "object",
+                        "additionalProperties": false,
+                        "properties": { "filesystem": { "type": "string" } },
```

Comment:
Filesystem as a target? To a storage non-expert like me this does not make sense. We should explain it in `title` or `description`

```json
                    "target": {
                        "filesystem": "..."
                    }                
```